### PR TITLE
config: allow to extend the module on Tarantool CE

### PR DESCRIPTION
**Pull request by Totktonada**
This is not a public API, at least now.

Part of #8862

### CODE REVIEW

**Review by ImeevMA — APPROVED** [review]
LGTM.

File: changelogs/unreleased/config-extras.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## feature/config
+
+* Allow to extend config's functionality using a non-public API on Tarantool
+  Community Edition (gh-8862).
```
Comment (p7nov) [inline changelogs/unreleased/config-extras.md:4]:
```suggestion
* Introduced a non-public API for extending the declarative config functionality
  in Tarantool Community Edition (gh-8862).
```

**Review by p7nov — APPROVED** [review]
Please see my suggestion on the changelog.

**Review by ylobankov — APPROVED** [review]
Tests LGTM

File: changelogs/unreleased/config-extras.md (hunk 1 above)
Comment (Totktonada) [reply changelogs/unreleased/config-extras.md]: Thanks! Updated.

**Pull request merged**